solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -1175,7 +1178,7 @@ private SolrCore(
       initSearcher(prev);
 
       // Initialize the RestManager
-      restManager = initRestManager();
+      restManager = isSynthetic() ? new RestManager() : initRestManager();
```

Comment:
how this is helping here?

Reply:
It was a quick prototype so I basically quickly handle NPEs when i ran into them. 

I believe for a better implementation, we will need to understand all the init happened in the `SolrCore`, there's a fair chance that we can simply bypass most of the init routine for a synthetic core (rest manager, event handler/listener, plugin, searcher etc), as synthetic core does not do any real searching, it simply relay it to the actual data node